trip_app/trips/models.py

```diff
@@ -34,16 +35,47 @@ class Trip(models.Model):
     num_seats = models.PositiveIntegerField()
     man_approve = models.BooleanField(default=True)
     description = models.TextField(blank=True)
-    is_active = models.BooleanField(default=True)
     created = models.DateTimeField(auto_now_add=True)
 
     @property
     def free_seats(self):
         """ Number of free seats that remained for the trip. """
         return self.num_seats - self.passengers.count()
 
+    def get_passengers(self):
+        """ String representation of trip passengers. """
+        return ",".join([str(p) for p in self.passengers.all()])
+
+    def process_request(self, user):
+        """ Create a trip request with active status if trip requires manual
+            approve, otherwise set approved status and add user to passenger's
+            list.
+
+            :raises: ValidationError when trip doesn't have free seats. """
+        if self.free_seats:
```

Comment:
For me it looks like, such validations should be derived to schema level.

Reply:
I have the same feeling, but I don't know how to implement this on the schema level. Basically the task is to limit the number of passengers (which is m2m field on Trip model) to the `num_seats`(field of a Trip model) number... The only solution I found is to use m2m signal which is calling handler function every time the ManyToManyField field is changed. I'm not sure if it is good enough though..Wondering what do you think about it. Please look at new updates.

Reply:
@Meathookseed  this is a good catch. 
@aleksandr-hilko maybe try the https://stackoverflow.com/a/36710502 approach?
After all the `process_request()` will take care of successful scenarios only.

Reply:
@andreybulygin potentially I can check the availability of free seats in the `validate() ` method of serializer. But I think there is one issue with this approach, because trip may have and may not have free seats and it seems hard to predict on the serializer level when this condition should be checked.

Reply:
@aleksandr-hilko sorry, I didn't quite get this point.